src/bootstrap_cache.rs

```diff
@@ -0,0 +1,163 @@
+use crate::dirs::Dirs;
+use crate::{Error, NodeInfo, R};
+use bincode::{deserialize_from, serialize_into};
+use log::{error, info};
+use std::fs::{self, File};
+use std::io::{self, BufReader, BufWriter};
+use std::path::Path;
+use std::path::PathBuf;
+
+/// Maximum peers in the cache.
+const MAX_CACHE_SIZE: usize = 200;
+
+type Stack = Vec<NodeInfo>;
```

Comment:
I wouldn't typdef this as it's used in just this file so seems like an overkill. But the main reason is that it's not really a stack - there's no LIFO semantics - in-fact quite the opposite with shuffling and all and also we are least likely to remove someone we added last as that's the freshest entry.